projects/hip-tests/catch/unit/device/hipGetDriverEntryPoint.cc

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANNTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER INN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR INN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <hip_test_common.hh>
+
+/**
+ * hipError_t hipGetDriverEntryPoint(const char* symbol, void** funcPtr, unsigned long long flags,
+                                   hipDriverEntryPointQueryResult* driverStatus);
+ * Gets function pointer of a request HIP API
+ *
+ * @param [in]  symbol  The API base name
+ * @param [out] funcPtr Pinter to the requested function
+ * @param [in]  flags Flags for the search
+ * @param [out] driverStatus Optional returned status of the search
+ *
+ * @returns #hipSuccess, #hipErrorInvalidValue
+ *
+ */
+
+/**
+ * Test Description
+ * ------------------------
+ *  - This will perfrom the funtionality testing of hipGetDriverEntryPoint api
+ * Test source
+ * ------------------------
+ *  - unit/device/hipGetDriverEntryPoint.cc
+ * Test requirements
+ * ------------------------
+ *  - HIP_VERSION >= 7.1
+ */
+TEST_CASE("Unit_hipGetDriverEntryPoint_Positive") {
+  void* funcPtr = nullptr;
+  hipDriverEntryPointQueryResult status;
+
+  SECTION("hipEnableDefault search flag") {
+    HIP_CHECK(hipGetDriverEntryPoint("hipGetDeviceCount", &funcPtr, hipEnableDefault, &status));
+  }
+
+  SECTION("hipEnableLegacyStream search flag") {
+    HIP_CHECK(
+        hipGetDriverEntryPoint("hipGetDeviceCount", &funcPtr, hipEnableLegacyStream, &status));
+  }
+
+  SECTION("hipEnablePerThreadDefaultStream search flag") {
+    HIP_CHECK(hipGetDriverEntryPoint("hipGetDeviceCount", &funcPtr, hipEnablePerThreadDefaultStream,
+                                     &status));
+  }
+
+  REQUIRE(status == hipDriverEntryPointSuccess);
+
+  hipError_t (*hipGetDeviceCount_ptr)(int*) = (hipError_t(*)(int*))funcPtr;
+  int countFuncPtr;
+  HIP_CHECK(hipGetDeviceCount_ptr(&countFuncPtr));
+
+  int count;
+  HIP_CHECK(hipGetDeviceCount(&count));
+  REQUIRE(countFuncPtr == count);
```

Comment:
Can we add one non zero check as well?
For some reason if it fails, they both will compare against failed values and it will pass.
Something like `REQUIRE(count > 0)` should help us.

Reply:
Its not addressed.